src/main/java/seedu/toluist/model/CommandAliasConfig.java

```diff
@@ -0,0 +1,111 @@
+package seedu.toluist.model;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * A model to manage alias
+ */
+public class CommandAliasConfig {
+    private static CommandAliasConfig instance;
+
+    private Map<String, String> aliasMapping = new HashMap<>();
+    private Set<String> reservedKeywords = new HashSet<>();
+
+    public Map<String, String> getAliasMapping() {
+        return aliasMapping;
+    }
+
+    public static CommandAliasConfig getInstance() {
+        if (instance == null) {
+            instance = new CommandAliasConfig();
+        }
+        return instance;
+    }
+
+    /**
+     * Set alias for a command. Returns success status
+     * @param alias
+     * @param command
+     * @return true / false
+     */
+    public boolean setAlias(String alias, String command) {
+        if (isReservedWord(alias)) {
+            return false;
+        }
+
+        aliasMapping.put(alias, command);
+        return true;
+    }
+
+    /**
+     * Remove an alias. Return sucess status
+     * @param alias
+     * @return true / false
```

Comment:
I thought it would be a ...
@guarantee alias will be removed and function will return true
?